client/helpers.go

```diff
@@ -121,7 +121,12 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 			return err
 		}
 		r.RemoveKeys(td.RemoveKeys)
-		r.RemovePaths(td.RemovePaths)
+
+		removePaths := td.RemovePaths
+		if td.ClearPaths {
```

Comment:
We could smartly remove paths (and it would be simpler code), but we would have to make it clear that there is smarter path logic around removing `""` to users, since we don't do any smart path logic anywhere else.  Users can currently add `""` and `path` as paths to the same delegation and both will display on the CLI, even though it doesn't really make sense in notary.

cc @endophage since we were talking earlier about how to manage paths via CLI